pkg/tsdb/influxdb/fsql/fsql.go

```diff
@@ -51,7 +53,21 @@ func Query(ctx context.Context, dsInfo *models.DatasourceInfo, req backend.Query
 		logger.Info(fmt.Sprintf("InfluxDB executing SQL: %s", qm.RawSQL))
 		info, err := r.client.Execute(ctx, qm.RawSQL)
 		if err != nil {
-			tRes.Responses[q.RefID] = backend.ErrDataResponse(backend.StatusInternal, fmt.Sprintf("flightsql: %s", err))
+			errStr := fmt.Sprintf("flightsql: %s", err)
```

Comment:
Nit so feel free to ignore
```suggestion
			errStr := fmt.Sprintf("flightsql: %s", err)
			tRes.Responses[q.RefID] = backend.ErrDataResponse(backend.StatusInternal, errStr)
			if grpcStatusErr, ok := status.FromError(err); ok {
				switch grpcStatusErr.Code() {
				case codes.InvalidArgument:
					tRes.Responses[q.RefID] = backend.ErrDataResponseWithSource(backend.StatusBadRequest, backend.ErrorSourceDownstream, errStr)
				case codes.PermissionDenied:
					tRes.Responses[q.RefID] = backend.ErrDataResponseWithSource(backend.StatusForbidden, backend.ErrorSourceDownstream, errStr)
				case codes.NotFound:
					tRes.Responses[q.RefID] = backend.ErrDataResponseWithSource(backend.StatusNotFound, backend.ErrorSourceDownstream, errStr)
				default:
					tRes.Responses[q.RefID] = backend.ErrDataResponse(backend.StatusInternal, errStr)
				}
			} 
```